src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -639,30 +639,10 @@ public virtual NamedTypeSymbol TupleUnderlyingType
         /// </remarks>
         internal abstract bool IsManagedType { get; }
 
-        //PROTOTYPE(span): this will completely change depending on how span-like types are implemented.
-        //                 Span<T> and ReadOnlySpan<T> will be special types (currently they are not always)
-        //                 Users may be able to define their own Spanlike types, but that is completely NYI 
-        //                 For now we will be simply looking at "System.Span" and "System.ReadOnlySpan" names
         /// <summary>
-        /// Returns true if the type is a Span/ReadOnlySpan
+        /// Returns true if the type may contain embedded references
         /// </summary>
-        internal bool IsSpanLikeType()
-        {
-            var originalDef = this.OriginalDefinition;
-
-            if (originalDef.Name != "Span" && originalDef.Name != "ReadonlySpan")
-            {
-                return false;
-            }
-
-            var ns = originalDef.ContainingSymbol as NamespaceSymbol;
-            if (ns?.Name != "System")
-            {
-                return false;
-            }
-
-            return (object)ns.ContainingNamespace != null;
-        }
+        internal virtual bool IsByRefLikeType => false;
```

Comment:
>internal virtual bool IsByRefLikeType => false; [](start = 8, length = 47)

APIs like this must be abstract so that we could ensure that all symbols have appropriate implementation.